### ISSUE TITLE
Segfault in CDC postimage

### ISSUE THREAD

**Issue by jul-stas** [scylladb/scylla#6143]
On fresh master, e.g. 4e6f5436767ef84072af7964b49fe56b0ffbc6c4 :
```
cqlsh> create table ksp.int2 (pk int, ck int, i1 int, i2 int, primary key (pk, ck)) with cdc={'enabled':true, 'preimage':true, 'postimage':true};
cqlsh> update ksp.int2 set i1=22 where pk=0 and ck=0;
NoHostAvailable:
```
(Segmentation fault on shard 1). Backtrace:
```
seastar::print_with_backtrace(seastar::backtrace_buffer&) at reactor.cc:?
seastar::print_with_backtrace(char const*) at reactor.cc:?
seastar::install_oneshot_signal_handler<11, &seastar::sigsegv_action>()::{lambda(int, siginfo_t*, void*)#1}::_FUN(int, siginfo_t*, void*) at reactor.cc:?
?? ??:0
cql3::untyped_result_set_row::has(seastar::basic_sstring<char, unsigned int, 15u, true> const&) const at untyped_result_set.cc:?
cdc::transformer::transform(mutation const&, cql3::untyped_result_set const*, long, seastar::basic_sstring<signed char, unsigned int, 31u, false>, int&)::{lambda(row const&, column_kind, clustering_key_prefix const&, std::optional<clustering_key_prefix>, cql3::untyped_result_set_row const*, std::optional<clustering_key_prefix>)#2}::operator()(row const&, column_kind, clustering_key_prefix const&, std::optional<clustering_key_prefix>, cql3::untyped_result_set_row const*, std::optional<clustering_key_prefix>) const at log.cc:?
cdc::transformer::transform(mutation const&, cql3::untyped_result_set const*, long, seastar::basic_sstring<signed char, unsigned int, 31u, false>, int&) at log.cc:?
seastar::noncopyable_function<seastar::future<> (seastar::lw_shared_ptr<cql3::untyped_result_set>&&)>::indirect_vtable_for<seastar::future<seastar::lw_shared_ptr<cql3::untyped_result_set> >::then<cdc::cdc_service::impl::augment_mutation_call(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >, std::vector<mutation, std::allocator<mutation> >&&, tracing::trace_state_ptr)::{lambda(std::vector<mutation, std::allocator<mutation> >&, service::query_state&, cdc::operation_details&)#2}::operator()(std::vector<mutation, std::allocator<mutation> >&, service::query_state&, cdc::operation_details&) const::{lambda(int)#1}::operator()(int)::{lambda(seastar::lw_shared_ptr<cql3::untyped_result_set>)#2}, seastar::future<> >(cdc::cdc_service::impl::augment_mutation_call(std::chrono::time_point<seastar::lowres_clock, std::chrono::duration<long, std::ratio<1l, 1000l> > >, std::vector<mutation, std::allocator<mutation> >&&, tracing::trace_state_ptr)::{lambda(std::vector<mutation, std::allocator<mutation> >&, service::query_state&, cdc::operation_details&)#2}::operator()(std::vector<mutation, std::allocator<mutation> >&, service::query_state&, cdc::operation_details&) const::{lambda(int)#1}::operator()(int)::{lambda(seastar::lw_shared_ptr<cql3::untyped_result_set>)#2}&&)::{lambda(seastar::lw_shared_ptr<cql3::untyped_result_set>&&)#1}>::call(seastar::noncopyable_function<seastar::future<> (seastar::lw_shared_ptr<cql3::untyped_result_set>&&)> const*, seastar::lw_shared_ptr<cql3::untyped_result_set>&&) at log.cc:?
seastar::continuation<seastar::internal::promise_base_with_type<>, seastar::future<seastar::lw_shared_ptr<cql3::untyped_result_set> >::then_impl_nrvo<seastar::noncopyable_function<seastar::future<> (seastar::lw_shared_ptr<cql3::untyped_result_set>&&)>, seastar::future<> >(seastar::noncopyable_function<seastar::future<> (seastar::lw_shared_ptr<cql3::untyped_result_set>&&)>&&)::{lambda()#1}::operator()() const::{lambda(seastar::internal::promise_base_with_type<>&, seastar::future_state<seastar::lw_shared_ptr<cql3::untyped_result_set> >&&)#1}, seastar::lw_shared_ptr<cql3::untyped_result_set> >::run_and_dispose() at log.cc:?
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at reactor.cc:?
seastar::reactor::run_some_tasks() at reactor.cc:?
seastar::reactor::run() at reactor.cc:?
seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::{lambda()#3}::operator()() const at reactor.cc:?
std::_Function_handler<void (), seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::{lambda()#3}>::_M_invoke(std::_Any_data const&) at reactor.cc:?
seastar::posix_thread::start_routine(void*) at posix.cc:?
```
Without `'postimage':true` issue does not appear.

**Comment by haaawk** [comment]
@avikivity This is a bug in 4.0 and it would be good to backport it to to 4.0

**Comment by nyh** [comment]
The fix is already in 4.1. Not backporting to 4.0 - although technically it's still supported (at least for a few more days...), CDC is still experimental and changing rapidly, and at this point, nobody should be using 4.0 with CDC.
